tart2ms/tart2ms.py

```diff
@@ -756,41 +948,53 @@ def __twelveball(direction):
             'OBSERVATION_ID': (("row",), da.from_array(np.zeros(row, dtype=int), chunks=chunks['row'])),
             'STATE_ID': (("row",), da.from_array(np.zeros(row, dtype=int), chunks=chunks['row'])),
         }
-        if fill_model and sources:
+        if fill_model:
             main_table["MODEL_DATA"] = (dims, model_data)
         dataset = Dataset(main_table)
         ms_datasets.append(dataset)
 
     ms_writes = xds_to_table(ms_datasets, ms_table_name, columns="ALL")
+    # auxilary table futures creation
     spw_writes = xds_to_table(spw_datasets, spw_table_name, columns="ALL")
     ddid_writes = xds_to_table(ddid_datasets, ddid_table_name, columns="ALL")
+    for tt in [ant_table, feed_table, field_table, pol_table, obs_table, src_table]:
+        tt.write()
 
-    dask.compute(ms_writes)
-
-    ant_table.write()
-    feed_table.write()
-    field_table.write()
-    pol_table.write()
-    obs_table.write()
-    src_table.write()
-
-    dask.compute(spw_writes)
-    dask.compute(ddid_writes)
```

Comment:
Very inefficient - rather put them into one compute call with a nice progress bar below once the graphs are all constructed.